embed/etcd.go

```diff
@@ -257,19 +257,21 @@ func startClientListeners(cfg *Config) (sctxs map[string]*serveCtx, err error) {
 		}
 
 		proto := "tcp"
+		addr := u.Host
 		if u.Scheme == "unix" || u.Scheme == "unixs" {
 			proto = "unix"
+			addr = u.Host + u.Path
```

Comment:
should you also update `sctxs` with `addr` too? https://github.com/coreos/etcd/blob/master/embed/etcd.go#L306